Marlin/ultralcd.cpp

```diff
@@ -3536,4 +3603,47 @@ void lcd_reset_alert_level() { lcd_status_message_level = 0; }
 
 #endif // ULTIPANEL
 
+#if ENABLED(ADC_KEYPAD)
+#define ADC_KEY_NUM     8
+typedef struct
+{
+    unsigned short ADCKeyValueMin;
+    unsigned short ADCKeyValueMax;
+    unsigned char  ADCKeyNo;
+}_stADCKeypadTable_;
+_stADCKeypadTable_ stADCKeyTable[ADC_KEY_NUM] =
+{
+    //VALUE_MIN, VALUE_MAX , KEY
+    { 2000,2048, BLEN_REPRAPWORLD_KEYPAD_F1 + 1 },      //F1
+    { 2000,2048, BLEN_REPRAPWORLD_KEYPAD_F2 + 1 },      //F2
+    { 2000,2048, BLEN_REPRAPWORLD_KEYPAD_F3 + 1 },      //F3
+    { 150,250,   BLEN_REPRAPWORLD_KEYPAD_LEFT + 1 },    //LEFT
+    { 950,1100,  BLEN_REPRAPWORLD_KEYPAD_RIGHT + 1 },   //RIGHT
+    { 280,440,   BLEN_REPRAPWORLD_KEYPAD_UP + 1 },      //UP
+    { 1340,1440, BLEN_REPRAPWORLD_KEYPAD_DOWN + 1 },    //DOWN
+    { 580,730,   BLEN_REPRAPWORLD_KEYPAD_MIDDLE + 1 },  //ENTER
+};
+
+  unsigned char get_ADC_keyValue(void)
+  {
+    if (thermalManager.ADCKey_count >= 16)
+    {
+      unsigned short currentkpADCValue = (thermalManager.current_ADCKey_raw / 8);
+      thermalManager.current_ADCKey_raw = 0;
+      thermalManager.ADCKey_count = 0;
+      if (currentkpADCValue < 1600)
+      {
+        for (unsigned char i = 0; i<ADC_KEY_NUM; i++)
+        {
+          if ((currentkpADCValue > stADCKeyTable[i].ADCKeyValueMin) && (currentkpADCValue < stADCKeyTable[i].ADCKeyValueMax))
+          {
+            return stADCKeyTable[i].ADCKeyNo;
+          }
+        }
+      }
+    }
+    return 0;
+  }
+#endif
+
```

Comment:
Please review again, just the point with "currentScreen ==" I am not sure with. It looks like to detect when to press and unpress a key, Maybe somebody with an A8 can check this.